### PR TITLE
Sell route

### PR DESCRIPTION
**Pull request by DrawnWren**
Fixed auth reducer on the front end and sellItem handler on the back end.

### CODE REVIEW

File: server/controllers/itemController.js [hunk 1]
```diff
@@ -45,22 +45,23 @@ module.exports = {
     });
   },
   sellItem(req, res, next) {
-    console.log(req);
-    console.log(req.user);
     const newItem = req.body;
+
+    const sellerId = (req.user ? req.user.user.id : 39);
```
Comment (JulianKnodt) [inline server/controllers/itemController.js]: What is this

File: server/controllers/itemController.js [hunk 2]
```diff
@@ -45,22 +45,23 @@ module.exports = {
...
+
     newItem.images = JSON.stringify(newItem.images);
     console.log('Creating new item,', newItem);
     db.items.create(newItem)
     .then(product => {
       db.items.getById(product[0])
       .then((result) => {
+        const id = result[0].id;
+        const transaction = { item_id: id, buyer_id: null, seller_id: sellerId };
```
Comment (JulianKnodt) [inline server/controllers/itemController.js]: I think you need to access the result's id through ['id']? if this works whatever

File: server/controllers/itemController.js (hunk 1 above)
Comment (DrawnWren) [inline server/controllers/itemController.js]: It says sellerId is either 39 or the actual user id

File: server/controllers/itemController.js (hunk 2 above)
Comment (DrawnWren) [inline server/controllers/itemController.js]: ['id'] and .id are the same. Property access via a . is actually just accessing a property with a string name.

File: client/app/reducers/auth.reducer.js [hunk 3]
```diff
@@ -7,18 +7,23 @@ const loginInit = {
...
+  const newState = state;
+
   if (dispatch === 'signin') {
-    state.user = action.user;
-    state.loggedIn = true;
-    return state;
+    newState.user = action.user;
+    newState.loggedIn = true;
+
+    return Object.assign({}, newState);
```
Comment (chrononaut) [inline client/app/reducers/auth.reducer.js]: Why not just do return Object.assign({}, state, {user: action.user, loggedIn: true}); ?

File: server/controllers/itemController.js (hunk 2 above)
Comment (JulianKnodt) [inline server/controllers/itemController.js]: I realize they have the same functionality, but in this case, I tried .id, and it somehow didn't work but the other one did

**Pull request merged**